src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -93,17 +94,19 @@ public static SamInputResource of(final File file) {
     /** Creates a {@link SamInputResource} reading from the provided resource, with no index. */
     public static SamInputResource of(final Path path) {
 
-        if (Files.isRegularFile(path) &&  Files.exists(path)) {
-            return new SamInputResource(new PathInputResource(path));
-        } else {
-            // in the case of named pipes and other non-seekable paths there's a bug in the implementation of
-            // java's GZIPInputStream which inappropriately uses .available() and then gets confused with the result
-            // of 0. For reference see:
-            // https://bugs.java.com/view_bug.do?bug_id=7036144
-            // https://github.com/samtools/htsjdk/pull/1077
-            // https://github.com/samtools/htsjdk/issues/898
+        // in the case of named pipes and other non-seekable paths there's a bug in the implementation of
+        // java's GZIPInputStream which inappropriately uses .available() and then gets confused with the result
+        // of 0. For reference see:
+        // https://bugs.java.com/view_bug.do?bug_id=7036144
+        // https://github.com/samtools/htsjdk/pull/1077
+        // https://github.com/samtools/htsjdk/issues/898
 
+        // This still doesn't support the case where someone is creating a named pipe in a non-default
+        // file system and then using it as input and passing a GZIPed into the other end of the pipe.
+        if (path.getFileSystem() == FileSystems.getDefault() && !Files.isRegularFile(path)) {
```

Comment:
Tested out this approach just now, and it seems to work fine for the cases we care about: no scheme, a `file://` scheme, and a recognized non-file scheme such as `gs://`. For unrecognized schemes like `fakescheme://` the expression `path.getFileSystem() == FileSystems.getDefault()` will evaluate to true -- but I imagine/hope that a downstream error will occur pretty soon afterwards in that case.